src/components/sidebar/right-sidebar/new-datepicker/components.tsx

```diff
@@ -0,0 +1,255 @@
+import * as React from "react";
+import { SliderItem } from "react-compound-slider";
+import './tooltip.css';
+
+const { Component, Fragment } = React;
+
+// *******************************************************
+// TOOLTIP RAIL
+// *******************************************************
+const railStyle: React.CSSProperties = {
+  position: "absolute",
+  width: "100%",
+  height: 40,
+  top: -13,
+  borderRadius: 7,
+  cursor: "pointer",
+  opacity: 0.3,
+  zIndex: 300,
+};
+
+const railCenterStyle: React.CSSProperties = {
+  position: "absolute",
+  width: "100%",
+  height: 5,
+  borderRadius: 7,
+  cursor: "pointer",
+  pointerEvents: "none",
+  backgroundColor: "#eaeaea"
+};
+
+interface TooltipRailProps {
+  activeHandleID: string;
+  getRailProps: (props: object) => object;
+  getEventData: (e: Event) => object;
+}
+
+export class TooltipRail extends Component<TooltipRailProps> {
```

Comment:
Probably easier to maintain in the long run if we just have each component in its own file here

Reply:
Might be unnecessary because all these components are from the same package. Just out of curiosity why would having an indiv file make it easier to maintain? (In the docx they have all comps in the same file - not that this is convention - but if we had to work on it together, would be easier to read the code if it was similar to the documentation)

Reply:
Consider the case where we know that the rail component specifically is buggy. Currently someone who wasn't super familiar with the code would have a difficult time figuring out where to start looking, since the component they would need to edit would be in a vaguely named `components.tsx` file. Discoverability would be better if it were in its own Rail.tsx file.

That being said, you guys are more familiar with the conventions here, so I'll leave the final judgement on what to do here up to you. Just a suggestion

Reply:
I agree with @atatmaja here, split up each component into its own file for improved maintainability in the future!

Reply:
Ahhhh yes that totally makes sense!